rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -260,13 +260,26 @@ private boolean validateLogsBloom(BlockHeader header, BlockResult result) {
         return Arrays.equals(calculateLogsBloom(result.getTransactionReceipts()), header.getLogsBloom());
     }
 
+    @VisibleForTesting
+    public BlockResult executeForMining(Block block, BlockHeader parent, boolean discardInvalidTxs) {
+        return executeForMining(block, parent, discardInvalidTxs, false);
+    }
+
+    public BlockResult executeForMining(Block block, BlockHeader parent, boolean discardInvalidTxs, boolean ignoreReadyToExecute) {
+        if (block.getHeader().getTxExecutionListsEdges() != null) {
+            return executeForMiningInternal(block, parent, discardInvalidTxs, ignoreReadyToExecute);
+        } else {
+            return executeOldSequential(null, 0, block, parent, discardInvalidTxs, ignoreReadyToExecute);
+        }
+    }
+
     @VisibleForTesting
     public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs) {
         return execute(block, parent, discardInvalidTxs, false);
     }
 
     public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs, boolean ignoreReadyToExecute) {
-        return executeInternal(null, 0, block, parent, discardInvalidTxs, ignoreReadyToExecute);
+        return executeParallelInternal(null, 0, block, parent, discardInvalidTxs, ignoreReadyToExecute);
     }
```

Comment:
Unless I missed something, these two methods can be removed and replaced for calls to `executeParallelInternal`, which can be public and renamed to simply `execute` because it doesn't always perform the execution in parallel.

Reply:
Right!